internal/paascharm/charmcraft_test.go

```diff
@@ -0,0 +1,104 @@
+package paascharm_test
+
+import (
+	"testing"
+
+	"github.com/kr/pretty"
+
+	"github.com/canonical/paascharmgen/internal/paascharm"
+)
+
+func TestGenerateGoStructsFromCharmcraftYaml(t *testing.T) {
+	yamlData := `
+name: go-app
+type: charm
+base: ubuntu@24.04
+platforms:
+  amd64:
+  arm64:
+  armhf:
+  ppc64el:
+  riscv64:
+  s390x:
+summary: A very short one-line summary of the Go application.
+description: |
+  A comprehensive overview of your Go application.
+extensions:
+  - go-framework
+config:
+  options:
+    user-defined-str:
+      type: string
+      default: "hello"
+      description: user-defined-str Description
+    user-defined-int:
+      type: int
+      default: 100
+      description: user-defined-int Description
+    user-defined-bool:
+      type: bool
+      description: user-defined-bool Description
+requires:
+  mysql:
+    interface: mysql_client
+    optional: true
+    limit: 1
+  s3:
+    interface: s3
+    optional: false
+
+parts: {0-git: {plugin: nil, build-packages: [git]}}
+`
+	packageName := "configpackage"
+	charmcraft, err := paascharm.ParseCharmcraftYaml([]byte(yamlData))
+	if err != nil {
+		t.Errorf("Error parsing charmcraft.yaml")
```

Comment:
You should always include the `err` in the test's error message, otherwise you'll have no idea what happened when it failed.
```suggestion
		t.Errorf("Error parsing charmcraft.yaml: %v", err)
```